### PR TITLE
backend: kubeconfig: Add test for missing context removal

### PR DESCRIPTION
**Pull request by CygnusMaximillian**
**Summary:**
Added backend tests to `kubeconfig` to improve test coverage and ensure robust error handling for edge cases in `RemoveContextFromFile`.

**Changes:**
- Added `TestRemoveContextFromFile_NonExistentContext` to verify the error returned when removing a context that does not exist in the kubeconfig file.
- Added `TestRemoveContextFromFile_EmptyPath` to verify that lock acquisition correctly fails and handles empty path strings.

**Steps to Test:**
1. Navigate to the backend directory: `cd backend`
2. Run the tests for the kubeconfig package: `npm run backend:test`
3. Verify that all tests pass and coverage is maintained or increased.

**Screenshots:**
*(Not applicable for backend test additions)*

**Notes for the Reviewer:**
These tests are small, strictly backward compatible, and improve the reliability of the `kubeconfig` file operations by explicitly covering previously untested error branches. No changes were made to the public API or existing logic.

### CODE REVIEW

File: backend/pkg/kubeconfig/file_test.go [hunk 1]
```diff
@@ -238,3 +238,25 @@ func TestRemoveContextFromFile(t *testing.T) {
...
+
+func TestRemoveContextFromFile_NonExistentContext(t *testing.T) {
+	data, err := os.ReadFile("./test_data/kubeconfig1")
+	require.NoError(t, err)
+	require.NotNil(t, data)
+
+	err = os.WriteFile("./test_data/config_copy_nonexistent", data, 0o600) //nolint:gosec
+	require.NoError(t, err)
+
+	defer os.Remove("./test_data/config_copy_nonexistent")
```
Comment (skoeva) [inline backend/pkg/kubeconfig/file_test.go]: could you address the linting errors here? you can see the linter results locally by running `make backend-lint-fix`

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for working on this.


Looks like the backend tests are failing in CI. Can you check with `cd backend && go test ./...` and fix the errors?

<details>
<summary>How to run the backend tests</summary>

Run `cd backend && go test ./...` to see all failures. Fix the failing tests and commit the result.

</details>

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for this PR.


The open review comments from Copilot still need attention — can you have a look? Once addressed, please mark them as resolved.

File: backend/pkg/kubeconfig/file_test.go (hunk 1 above)
Comment (CygnusMaximillian) [reply backend/pkg/kubeconfig/file_test.go]: sure

**Comment by CygnusMaximillian** [comment]
@illume all the issues have been resolved you can take a look now

**Comment by CygnusMaximillian** [comment]
also the copilot comments are also resolved

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for the contribution.


Could you take a look at the commit messages in this PR? We follow a Linux kernel style for git commits — see the [contributing guide](https://headlamp.dev/docs/latest/contributing#2-follow-commit-guidelines) and `git log` for examples.

<details>
<summary>Commits that need attention</summary>

- `backend: kubeconfig: resolve the linting error and use require instead of assert for checking` — Description must start with a capital letter — e.g. `frontend: HomeButton: Fix the button` not `frontend: HomeButton: fix the button`.

</details>

<details>
<summary>Commit guidelines</summary>

- Use atomic commits focused on a single change.
- Use the title format `<area>: <Description of changes>` — description must start with a capital letter.
- Keep the title under 72 characters (soft requirement).
- Explain the intention and why the change is needed.
- Make commit titles meaningful and describe what changed.
- Do not add code that a later commit rewrites; squash or reorder commits instead.
- Do not include `Fixes #NN` in commit messages.

Good examples:

- `frontend: HomeButton: Fix so it navigates to home`
- `backend: config: Add enable-dynamic-clusters flag`

</details>

**Comment by illume** [comment]
I'll squash the commits.

**Review by illume — APPROVED** [review]
🎉 thanks!
